File: ios/SneakBit/Controller/EmulatedKey.swift
import Foundation

enum EmulatedKey {
    case up
    case right
    case down
    case left
    case closeRangeAttack
    case rangedAttack
    case backspace
    case confirm
    case escape
    case menu
}

extension EmulatedKey {
    var imageName: String {
        switch self {
        case .up: "up"
        case .down: "down"
        case .left: "left"
        case .right: "right"
        case .closeRangeAttack: "close_attack"
        case .rangedAttack: "attack"
        case .backspace: ""
        case .confirm: "confirm"
        case .escape: ""
        case .menu: "menu"
        }
    }
}

extension EmulatedKey {
    var isMovement: Bool {
        switch self {
        case .up, .right, .down, .left: true
        default: false
        }
    }
}

File: android/app/src/main/java/it/curzel/bitscape/controller/EmulatedKey.kt
package it.curzel.bitscape.controller

import androidx.annotation.DrawableRes
import it.curzel.bitscape.R

enum class EmulatedKey {
    UP,
    RIGHT,
    DOWN,
    LEFT,
    CLOSE_RANGE_ATTACK,
    RANGED_ATTACK,
    BACKSPACE,
    CONFIRM,
    ESCAPE,
    MENU;

    @get:DrawableRes
    val imageKeyUp: Int get() = when (this) {
        CLOSE_RANGE_ATTACK -> R.drawable.close_attack_button_up
        RANGED_ATTACK -> R.drawable.kunai_button_up
        CONFIRM -> R.drawable.confirm_button_up
        else -> 0
    }

    @get:DrawableRes
    val imageKeyDown: Int get() = when (this) {
        CLOSE_RANGE_ATTACK -> R.drawable.close_attack_button_down
        RANGED_ATTACK -> R.drawable.kunai_button_down
        CONFIRM -> R.drawable.confirm_button_down
        else -> 0
    }
}
